src/rnn/attention.h

```diff
@@ -44,18 +60,55 @@ class GlobalAttention : public CellInput {
     dropout_ = options_->get<float>("dropout", 0);
     layerNorm_ = options_->get<bool>("layer-normalization", false);
     nematusNorm_ = options_->get<bool>("nematus-normalization", false);
+    numAttentionHeads_ = options_->get<int>("attentionHeads", 1);
+    attentionLookupDim_ = options_->get<int>("attentionLookupDim", -1);
+    attentionProjectionDim_ = options_->get<int>("attentionProjectionDim", -1);
+    attentionIndependentHeads_ = options->get<bool>("attentionIndependentHeads", false);
+    attentionBilinearLookup_ = options->get<bool>("attentionBilinearLookup", false);
+    attentionProjectionLayerNorm_ = options->get<bool>("attentionProjectionLayerNorm", false);
+    attentionProjectionTanH_ = options->get<bool>("attentionProjectionTanH", false);
     std::string prefix = options_->get<std::string>("prefix");
+    //LOG(info, "Attention heads: {}", numAttentionHeads_);
 
     int dimEncState = encState_->getContext()->shape()[-1];
 
-    Wa_ = graph->param(prefix + "_W_comb_att",
-                       {dimDecState, dimEncState},
-                       inits::glorot_uniform);
-    Ua_ = graph->param(
-        prefix + "_Wc_att", {dimEncState, dimEncState}, inits::glorot_uniform);
-    va_ = graph->param(
-        prefix + "_U_att", {dimEncState, 1}, inits::glorot_uniform);
-    ba_ = graph->param(prefix + "_b_att", {1, dimEncState}, inits::zeros);
+    attentionLookupDim_ = (attentionLookupDim_ == -1)? dimEncState: attentionLookupDim_;
+    attentionIndependentHeads_ = (attentionBilinearLookup_)? true: attentionIndependentHeads_;
+    for (int headI = 0; headI < numAttentionHeads_; ++headI) {
```

Comment:
Further up you used `headId`. Is that the same? Please use the same name for the same things.

Reply:
In this file I've used headI as a for iteration index, which is a different usage than headId in s2s.h